flake.nix

```diff
@@ -11,126 +11,65 @@
         "x86_64-darwin"
         "x86_64-linux"
       ];
-      foreachSystem = f: lib.foldl' (attrs: system: lib.recursiveUpdate attrs (f system)) { } systems;
+      foreachSystem = lib.genAttrs systems;
     in
     {
-      overlay = (final: prev: {
-        git-branchless = final.callPackage
-          (
-            { lib
-            , git
-            , libiconv
-            , ncurses
-            , openssl
-            , pkg-config
-            , rustPlatform
-            , sqlite
-            , stdenv
-            , Security
-            , SystemConfiguration
-            }:
+      overlays.default = (final: prev: {
 
-            rustPlatform.buildRustPackage {
-              name = "git-branchless";
-
-              src = self;
-
-              cargoLock = {
-                lockFile = "${self}/Cargo.lock";
-              };
-
-              nativeBuildInputs = [ pkg-config ];
-
-              buildInputs = [
-                ncurses
-                openssl
-                sqlite
-              ] ++ lib.optionals stdenv.isDarwin [
-                Security
-                SystemConfiguration
-                libiconv
-              ];
-
-              preCheck = ''
-                export TEST_GIT=${git}/bin/git
-                export TEST_GIT_EXEC_PATH=$(${git}/bin/git --exec-path)
-              '';
-              # FIXME: these tests time out when run in the Nix sandbox
-              checkFlags = [
-                "--skip=test_switch_pty"
-                "--skip=test_next_ambiguous_interactive"
-                "--skip=test_switch_auto_switch_interactive"
-              ];
-            }
-          )
-          {
-            inherit (final.darwin.apple_sdk.frameworks) Security SystemConfiguration;
+        # reuse the definition from nixpkgs git-branchless
+        git-branchless = prev.git-branchless.overrideAttrs ({ meta, ... }: {
```

Comment:
Back when I contributed the Flake I consciously decided against this approach because it will be cumbersome to add extra dependencies or disable test that fail in the Nix sandbox. However, if you believe that the package definition has converged sufficiently, please don't feel any obligation.

Reply:
I also have this concern! Though I do think that currently things have stabilized enough to allow us to do this. I would like to contribute to both nixpkgs and git-branchless in the long term (this is really an awesome project), and if things got complicated again, it would then be better to bring back the fully local package definition. But for now, I think overrides are good enough, and this eliminates the need to port things back and forth (between nixpkgs and flake.nix here).

Reply:
I will mark this as resolved for now, but if you have any other comments, please feel free to unresolve any time!